File: asp-dot-net/Models/Transaction.cs
public class Transaction
{
    public int id { get; set; }
    public string? transaction_id { get; set; }
    public string? merchant { get; set; }
    public string? mcc { get; set; }
    public string? currency { get; set; }
    public double amount { get; set; }
    public string? transaction_date { get; set; }
    public string? card_id { get; set; }
    public string? card_pan { get; set; }
    public string? card_type { get; set; }
}

File: go-gin/models/transaction.go
package models

import "mime/multipart"

type TransactionFile struct {
	File *multipart.FileHeader `form:"file" binding:"required"`
}

type TransactionObj struct {
	Id              int     `json:"id"`
	TransactionId   string  `json:"transaction_id"`
	Merchant        string  `json:"merchant"`
	MCC             string  `json:"mcc"`
	Currency        string  `json:"currency"`
	Amount          float32 `json:"amount"`
	TransactionDate string  `json:"transaction_date"`
	CardId          string  `json:"card_id"`
	CardPan         string  `json:"card_pan"`
	CardType        string  `json:"card_type"`
}

type TransactionRow struct { // Our example struct, you can use "-" to ignore a field
	Id              int     `csv:"id"`
	TransactionId   string  `csv:"transaction_id"`
	Merchant        string  `csv:"merchant"`
	MCC             string  `csv:"mcc"`
	Currency        string  `csv:"currency"`
	Amount          float32 `csv:"amount"`
	TransactionDate string  `csv:"transaction_date"`
	CardId          string  `csv:"card_id"`
	CardPan         string  `csv:"card_pan"`
	CardType        string  `csv:"card_type"`
}
